src/util/format_spec.h

```diff
@@ -16,12 +16,12 @@ class format_spect
   stylet style;
 
   format_spect()
-    : min_width(0), precision(6), zero_padding(false), style(AUTOMATIC)
+    : min_width(0), precision(16), zero_padding(false), style(AUTOMATIC)
```

Comment:
Do we need to implement this change to make the witness validators happy?

Reply:
It's in order to have the bit-pattern match how C would parse the decimal floating point numbers. If C gets a different bit pattern, all bets are off that the counter-example actually is one. So, yes and no: I needed this when testing validators to ensure they actually get what ESBMC computed.

So for witness files this should surely be done, but maybe not with the current approach. I would argue that also for counter-examples we should start using one of the existing algorithms that produce the shortest unique float representation.

Reply:
For instance the `State 5` in #1492 is currently represented to the user as
```
  y = 2.425029e-1f (00111110 01111000 01010010 10110000)
```
But parsing that via strtod() or strtof() yields `0x 3e 78 52 ae`, so the last 5 bits changed.